### PR TITLE
Add support for Tensorflow SparseTensors: core layers.

### PR DESCRIPTION
**Pull request by hertschuh**
Added `tf.SparseTensor` support for ops:
- matmul
- take

Added `tf.SparseTensor` support for core layers:
- Dense
- Embedding
- Identity

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/backend/tensorflow/numpy.py [hunk 1]
```diff
@@ -550,6 +568,28 @@ def swapaxes(x, axis1, axis2):
 
 
 def take(x, indices, axis=None):
+    if isinstance(indices, tf.SparseTensor):
+        if x.dtype not in (tf.float16, tf.float32, tf.float64, tf.bfloat16):
+            warnings.warn(
+                "`keras_core.backend.tensorflow.numpy.take` does not support "
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras_core/backend/tensorflow/numpy.py]: Done

File: keras_core/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -550,6 +568,28 @@ def swapaxes(x, axis1, axis2):
...
+                f"`x.dtype={x.dtype}` when `indices` is a sparse tensor; "
+                "densifying `indices`."
+            )
+            return tfnp.take(x, tf.sparse.to_dense(indices), axis=axis)
+        if axis is None:
+            x = tf.reshape(x, (-1,))
+        elif axis != 0:
+            warnings.warn(
+                "`keras_core.backend.tensorflow.numpy.take` does not support "
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply keras_core/backend/tensorflow/numpy.py]: Done

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**